cinn/hlir/pe/new_schedule.cc

```diff
@@ -0,0 +1,543 @@
+    auto loops      = ir_sch.GetLoops(all_blocks[0]);
+    int last_shape  = ir::GetLoopExtent(loops[dims - 1]);
+    factor          = GetVectorizeFactor(last_shape, factor);
+    auto splited    = ir_sch.Split(loops[dims - 1], {-1, factor});
+    ir_sch.Vectorize(splited[1], factor);
+    if (dims == 1) {
+      ir_sch.Parallel(splited[0]);
+    }
+  }
+}
+
+void NewCudaScheduleInjective(ir::IRSchedule &ir_sch,
+                              const std::vector<int> &output_shape,
+                              const common::Target &target) {
+  auto all_blocks = ir_sch.GetAllBlocks();
+  auto loops      = ir_sch.GetLoops(all_blocks[0]);
+  auto fused      = ir_sch.Fuse(loops);
+
+  int num_thread        = target.max_num_threads();
+  int num_block         = 1024;
+  int vector_width      = 1;
+  int prod_size         = std::accumulate(output_shape.begin(), output_shape.end(), 1, std::multiplies<int>());
+  bool need_block_split = prod_size > num_thread * num_block * vector_width ? true : false;
+  if (need_block_split) {
+    auto splited = ir_sch.Split(fused, {num_block, num_thread, -1});
+    ir_sch.Bind(splited[0], "blockIdx.x");
+    ir_sch.Bind(splited[1], "threadIdx.x");
+  } else {
+    if (prod_size > num_thread) {
+      auto splited = ir_sch.Split(fused, {-1, num_thread});
+      ir_sch.Bind(splited[0], "blockIdx.x");
+      ir_sch.Bind(splited[1], "threadIdx.x");
+    } else {
+      ir_sch.Bind(fused, "threadIdx.x");
+    }
+  }
+  LOG(INFO) << "After NewCudaScheduleInjective, new ir is : " << ir_sch.GetModule().GetExprs().at(0);
+}
+
+void NewCudaScheduleMul(ir::IRSchedule &ir_sch, const std::vector<int> &output_shape, const common::Target &target) {
+  auto all_blocks = ir_sch.GetAllBlocks();
+  auto loops      = ir_sch.GetLoops(all_blocks.back());
+  auto splited    = ir_sch.Split(loops[1], {-1, 2});
+  all_blocks      = ir_sch.GetAllBlocks();
+  loops           = ir_sch.GetLoops(all_blocks.back());
+  ir_sch.Bind(loops[0], "blockIdx.x");
+  ir_sch.Bind(loops[1], "threadIdx.x");
+}
+
+void NewMulScheduleCPU(ir::IRSchedule &ir_sch,
+                       const std::vector<int> &reduce_first_shape,
+                       const common::Target &target) {
+  ir_sch.MergeExprs();
+  auto all_blocks = ir_sch.GetAllBlocks();
+  CHECK_EQ(all_blocks.size(), 4U);
+  auto loops    = ir_sch.GetLoops(all_blocks[1]);
+  int loop_size = loops.size();
+  // ir_sch.Reorder({loops[loop_size-1], loops[loop_size-2]});
+
+  if (reduce_first_shape.back() > 1) {
+    all_blocks = ir_sch.GetAllBlocks();
+    loops      = ir_sch.GetLoops(all_blocks[3]);
+    ir_sch.Unroll(loops.back());
+  }
+}
+
+void NewCudaSplitSchedule(ir::IRSchedule &ir_sch,
+                          const std::vector<std::vector<int>> &output_shapes,
+                          int axis,
+                          const common::Target &target) {
+  ir_sch.MergeExprs();
+  int dims = output_shapes[0].size();
+  std::vector<int> reorders;
+  for (int i = 0; i < dims; i++) {
```

Comment:
Prefer `++i` in the future, reference:

https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement

Reply:
Done, thanks